common/src/main/java/com/wynntils/services/mapdata/providers/builtin/PlayerProvider.java

```diff
@@ -0,0 +1,266 @@
+        reloadHadesUsers();
+    }
+
+    private void reloadHadesUsers() {
+        PLAYERS.forEach(this::notifyCallbacks);
+        PLAYERS.clear();
+        Services.Hades.getHadesUsers().map(RemotePlayerLocation::new).forEach(PLAYERS::add);
+    }
+
+    @Override
+    public String getProviderId() {
+        return "players";
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        return PLAYERS.stream().map(MapFeature.class::cast);
+    }
+
+    @Override
+    public Stream<MapCategory> getCategories() {
+        return PROVIDED_CATEGORIES.stream();
+    }
+
+    private static final class PlayersCategory implements MapCategory {
+        @Override
+        public String getCategoryId() {
+            return "wynntils:player";
+        }
+
+        @Override
+        public Optional<String> getName() {
+            return Optional.of("Player positions");
+        }
+
+        @Override
+        public Optional<MapAttributes> getAttributes() {
+            return Optional.of(new AbstractMapAttributes() {
+                @Override
+                public Optional<Integer> getPriority() {
+                    return Optional.of(900);
+                }
+
+                @Override
+                public Optional<MapVisibility> getIconVisibility() {
+                    return Optional.of(FixedMapVisibility.ICON_ALWAYS);
+                }
+            });
+        }
+    }
+
+    public static final class RemotePlayerLocation implements MapLocation {
+        private final HadesUser hadesUser;
+
+        public RemotePlayerLocation(HadesUser hadesUser) {
+            this.hadesUser = hadesUser;
+        }
+
+        @Override
+        public Location getLocation() {
+            return new Location((int) hadesUser.getX(), (int) hadesUser.getY(), (int) hadesUser.getZ());
+        }
+
+        @Override
+        public String getFeatureId() {
+            return hadesUser.getName().toLowerCase(Locale.ROOT);
+        }
+
+        @Override
+        public String getCategoryId() {
+            return "wynntils:players:" + (hadesUser.getRelation().isEmpty() ? "other" : hadesUser.getRelation());
+        }
+
+        @Override
+        public Optional<MapAttributes> getAttributes() {
+            return Optional.of(new AbstractMapAttributes() {
+                @Override
+                public Optional<String> getLabel() {
+                    return Optional.of(hadesUser.getName());
+                }
+
+                @Override
+                public Optional<CustomColor> getLabelColor() {
+                    return Optional.of(hadesUser.getRelationColor());
+                }
+
+                @Override
+                public Optional<String> getIconId() {
+                    return Optional.of("wynntils:icon:player:head");
+                }
+
+                @Override
+                public Optional<MapDecoration> getIconDecoration() {
+                    return Optional.of(new MainMapIconDecoration());
```

Comment:
I think I'd prefer to make `MainMapIconDecoration` static and pass the hadesUser to the constructor instead. Using nested classes with an implicit parent this is tricky, I think, and should only be used if it is the only possible solution.

Also perhaps rename the class to `PlayerMapDecoration`?

Reply:
... in fact, seeing how this is used, maybe we should even rename the class from `MapDecoration` to `MapDecorator`? If you agree, we can do that in a follow-up change, so we don't mix more changes into this PR.

Reply:
> ... in fact, seeing how this is used, maybe we should even rename the class from `MapDecoration` to `MapDecorator`? If you agree, we can do that in a follow-up change, so we don't mix more changes into this PR.

Sure, I don't mind either way.